devito/types/basic.py

```diff
@@ -550,12 +667,49 @@ def name(self):
     @property
     def indices(self):
         """The indices (aka dimensions) of the object."""
-        return self._indices
+        return self.args
+
+    @property
+    def index_ref(self):
+        """The indices (aka dimensions) of the object."""
+        return self._index_ref
+
+    @property
+    def ind_map(self):
+        return {d1: d2 for d1, d2 in zip(self.dimensions, self.index_ref)}
+
+    @property
+    def origin(self):
```

Comment:
not that this `origin` is different than `Grid.origin` ... perhaps not ideal...

Reply:
This is the origin of the staggered grid, which is either the grid stores the originis of each possible staggering and related properties, or a staggered function stores it itself which makes it a lot less messy

Reply:
you should have `self.dimensions` on the LHS of the `zip`, for consistency with other properties